docs(section): Add section on Kubernetes

ishijo opened the pull request:
This pull request adds documentation for Kubernetes components and Helm packaging.

Changes:
- Kubernetes core concepts (Pods, Deployments, Services, ConfigMaps, Secrets, Namespaces, PVs)
- Kubernetes architecture overview
- Helm charts structure and examples
- Common Helm commands and best practices

Closes #61

ishijo committed:
docs(section): Add section on Kubernetes

lsetiawan reviewed (changes requested):
Hi @ishijo Thanks for this PR. Looks great so far. However, there are some formatting issues that cause rendering on the site, to not be quite right. Please see https://rse-guidelines--95.org.readthedocs.build/en/95/advanced/deployment-strategies-tools/kubernetes/
<img width="1066" height="528" alt="Screenshot 2025-10-27 at 9 40 38 AM" src="https://github.com/user-attachments/assets/4aff3e6a-e011-4ef9-98b6-436f3996aafb" />

ishijo committed:
fixed rendering issue

ishijo committed:
Merge remote-tracking branch 'upstream/main' into kubernetes_section

ishijo committed:
docs(kubernetes): address review feedback from #104

- fix bullet list rendering (blank line before lists, remove trailing hard breaks)
- remove {% raw %} wrapper (mkdocs escapes {{ }} in fenced code by default)
- update container runtime examples to containerd, CRI-O
- correct master-worker to control plane / node terminology
- remove 404'd Red Hat ebook link
- add base64 encoding caveat to Secret example
- add trailing newline

Closes #104
